packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -794,13 +688,15 @@ class SyncWatcher extends BaseWatcher {
       return
     }
     if (!sourceChainId) {
-      logger.error('expected sourceChainId')
+      logger.warn(`populateTransferSentTimestamp marking item not found: sourceChainId. dbItem: ${JSON.stringify(dbTransfer)}`)
+      await this.db.transfers.update(transferId, { isNotFound: true })
       return
     }
     const sourceBridge = this.getSiblingWatcherByChainId(sourceChainId).bridge // eslint-disable-line @typescript-eslint/no-non-null-assertion
     const timestamp = await sourceBridge.getBlockTimestamp(transferSentBlockNumber)
     if (!timestamp) {
-      logger.warn(`timestamp not found for block number ${transferSentBlockNumber} on sourceChainId ${sourceChainId}`)
+      logger.warn(`populateTransferSentTimestamp marking item not found: timestamp for block number ${transferSentBlockNumber} on sourceChainId ${sourceChainId}. dbItem: ${JSON.stringify(dbTransfer)}`)
+      await this.db.transfers.update(transferId, { isNotFound: true })
```

Comment:
Do you think it would be possible for getBlockTimestamp here to return null if the rpc provider hasn't yet indexed the block yet? basically would it be a possibility to prematurely set this item as not found because it couldn't find the block info but it eventually does exist (I think we saw some endpoints in the past not return info right away)

Reply:
This was true when there was a `0` block delay for ORUs AND we were using the ORU RPC provider instead of an aggregated node service (Infura or Alchemy). There is now a delay and the node services we use aggregate data.

Because of both of those facts, I believe we will not ever prematurely set this as not found.